heroic-core/src/main/java/com/spotify/heroic/CoreQueryManager.java

```diff
@@ -232,8 +240,17 @@ public Aggregation visitString(final StringExpression e) {
 
                 final OptionalLimit limit = options.getGroupLimit().orElse(groupLimit);
 
-                return async.collect(futures,
+                AsyncFuture<QueryResult> queryResult = async.collect(futures,
                     QueryResult.collectParts(QUERY, range, combiner, limit));
+
+                if (shouldLogQueries) {
+                    queryResult = queryResult.directTransform(_queryResult -> {
```

Comment:
We generally don't use any underscore convention.

Also, no need to transform the result, do this instead:
```java
queryResult.onResolved(result -> {
  queryLogger.logQueryResolved(q, result);
});
```